src/snowflake/connector/auth/by_plugin.py

```diff
@@ -207,20 +191,22 @@ def handle_timeout(
         time ranges between 1 and 16 seconds.
         """
 
-        del authenticator, service_name, account, user, password
+        if delete_params:
+            del authenticator, service_name, account, user, password
+
         logger.debug("Default timeout handler invoked for authenticator")
         if not self._retry_ctx.should_retry():
             error = OperationalError(
-                msg=f"Could not connect to Snowflake backend after {self._retry_ctx.get_current_retry_count()} attempt(s)."
+                msg=f"Could not connect to Snowflake backend after {self._retry_ctx.current_retry_count + 1} attempt(s)."
```

Comment:
As is, `current_retry_count` tracks the number of retries that has already happened, so total attempts is the number of retries +1. I could fiddle around with `retry_ctx` to make it so that we don't need this +1 here if necessary.

A note here is that the old code (without the +1) was simply wrong. After the first failed connection attempt it would say "could not connect... after 0 attempts".

However, we can also change the message to `Could not connect to Snowflake backend after X retrie(s)` to get rid of the +1, which is much easier.